### ISSUE TITLE
Errors in homotopy implementations in MSL 3.2.1

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#974]
**Modified by dietmarw on 28 Jan 2013 09:46 UTC**
It seems that there are errors in homotopy implementations in MSL 3.2.1:

 1. in Modelica.Fluid.Fittings.BaseClasses.QuadraticTurbulent.BaseModel:
 ```mo
  if from_dp then
     m_flow = homotopy(...,
                      m_flow_nominal*dp_fg/dp_nominal);
  else
     dp_fg = homotopy(...,
                      m_flow_nominal*dp_fg/dp_nominal);
  end if;
```
 The second expression should be something like: simplified=**dp_nominal\*m_flow/m_flow_nominal**

 2. in Modelica.Fluid.Fittings.SimpleGenericOrifice:
 ```mo
  if from_dp then
    m_flow = homotopy(...,
                       m_flow_nominal*dp_fg/dp_nominal);
  else
    dp_fg = homotopy(...,
                     m_flow_nominal*dp_fg/dp_nominal);
  end if;
```
 The second expression should be something like: simplified=**dp_nominal\*m_flow/m_flow_nominal**

 3. MSL 3.3Dev is built completely differently, and doesn't contain these models.

Best regards,

Eric Thomas 

----

**Reported by Eric Thomas on 27 Jan 2013 19:13 UTC**
It seems that there are errors in homotopy implementations in MSL 3.2.1:

1/ in Modelica.Fluid.Fittings.BaseClasses.QuadraticTurbulent.BaseModel:

  if from_dp then
     m_flow = homotopy(...,
                      m_flow_nominal\*dp_fg/dp_nominal);
  else
     dp_fg = homotopy(...,
                      **m_flow_nominal\*dp_fg/dp_nominal**);
  end if;

The second expression should be something like: simplified=**dp_nominal\*m_flow/m_flow_nominal**

2/ in Modelica.Fluid.Fittings.SimpleGenericOrifice:

  if from_dp then
    m_flow = homotopy(...,
                       m_flow_nominal\*dp_fg/dp_nominal);
  else
    dp_fg = homotopy(...,
                     **m_flow_nominal\*dp_fg/dp_nominal**);
  end if;

The second expression should be something like: simplified=**dp_nominal\*m_flow/m_flow_nominal**

3/ MSL 3.3Dev is built completely differently, and doesn't contain these models.

Best regards,

Eric Thomas 

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/974

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 27 Jan 2013 19:16 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 28 Jan 2013 09:46 UTC**
Improved readability of code listing.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 28 Jan 2013 09:46 UTC**
Thomas, I'm wondering which MSL 3.3Dev you are referring to. Currently there is no such thing and in `trunk` is the development version that is going to be released as MSL 3.2.1. Could you clarify?

**Comment by modelica-trac-importer** [comment]
**Comment by eth on 30 Jan 2013 21:32 UTC**
You are right. It seems that I have confused several development versions. I don't find it again. It is better.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 30 Jan 2013 21:46 UTC**
Now I have to ask again:
 1. Did you find this issue in the Specification or in the Modelica Standard Library? 
 2. Is the issue still present?

**Comment by modelica-trac-importer** [comment]
**Comment by rfranke on 31 Jan 2013 18:06 UTC**
the bug is present in the trunk and needs to be solved for MSL 3.2.1.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 31 Jan 2013 18:07 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 23 Feb 2013 12:57 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by fcasella on 5 Mar 2013 22:38 UTC**
Fixed in e888ba44b4688768758900269d9791864ae10461
